addons/controllers/csi/vspherecsiconfig_controller.go

```diff
@@ -319,6 +324,10 @@ func (r *VSphereCSIConfigReconciler) ConfigMapToVSphereCSIConfig(o client.Object
 	_ = r.List(context.Background(), configs)
 	requests := []ctrl.Request{}
 	for i := 0; i < len(configs.Items); i++ {
+		// skip template vSphereCSIConfig CRs
+		if _, ok := configs.Items[i].Annotations[constants.TKGAnnotationTemplateConfig]; ok {
```

Comment:
**question**: Why does vsphereCSI handler alone need this change ?

Reply:
Just saw the note in the PR about this. Looks good. I thought predicates filtering is done before calling handler functions. Seems like thats not the case.

Reply:
Thanks for the review! Yeah, the controller also watches create/update events on related `ConfigMap` CR to reconcile `VSphereCSIConfig` CRs. I will add more comments here to explain why we still need an annotation check.